test/lint/lint-git-commit-check.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+# Copyright (c) 2020 The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+#
+# Linter to check that commit messages have a new line before the body
+# or no body at all
+
+export LC_ALL=C
+
+EXIT_CODE=0
+
+while getopts "?" opt; do
+  case $opt in
+    ?)
+      echo "Usage: $0 [N]"
+      echo "       TRAVIS_COMMIT_RANGE='<commit range>' $0"
+      echo "       $0 -?"
+      echo "Checks unmerged commits, the previous N commits, or a commit range."
+      echo "TRAVIS_COMMIT_RANGE='47ba2c3...ee50c9e' $0"
+      exit ${EXIT_CODE}
+    ;;
+  esac
+done
+
+if [ -z "${TRAVIS_COMMIT_RANGE}" ]; then
+  if [ -n "$1" ]; then
+    TRAVIS_COMMIT_RANGE="HEAD~$1...HEAD"
+  else
+    TRAVIS_COMMIT_RANGE="origin/master..HEAD"
+  fi
+fi
+
+while IFS= read -r commit_hash  || [[ -n "$commit_hash" ]]; do
+    n_line=0
+    while IFS= read -r line || [[ -n "$line" ]]; do
+        n_line=$((n_line+1))
+        length=${#line}
+        if [ $n_line -eq 2 ] && [ $length -ne 0 ]; then
+            echo "The body of commit hash ${commit_hash} is followed by a non-empty line. Subject lines should always be followed by a blank line."
```

Comment:
Good idea adding the commit hash!  The grammar still isn't quite what I'd like.

```suggestion
            echo "The subject line of commit hash ${commit_hash} is followed by a non-empty line. Subject lines should always be followed by a blank line."
```
or, if you like it better
```suggestion
            echo "The subject line of commit hash ${commit_hash} is followed by a non-empty line. Please add a blank line between the subject line and the body of the commit message."
```